src/main.cpp

```diff
@@ -4436,23 +4436,26 @@ void GridcoinServices()
     if (TimerMain("gather_cpids",480))
         msNeuralResponse.clear();
 
-    if (TimerMain("check_for_autoupgrade",240))
+#ifdef QT_GUI
+    // Check for updates once per day.
+    if(GetAdjustedTime() - nLastCheckedForUpdate > 24 * 60 * 60)
```

Comment:
so this would essentially also check for upgrade at start of GridcoinServices when wallet is loaded if they opt in?

Reply:
Yup, that's right. Once on the first block received/staked, then every 24h.

Reply:
not bad idea. and i agree opt in should be required.